app/code/Magento/Swatches/Model/Plugin/EavAttribute.php

```diff
@@ -174,8 +174,10 @@ protected function prepareOptionIds(array $optionsArray)
     {
         if (isset($optionsArray['value']) && is_array($optionsArray['value'])) {
             foreach (array_keys($optionsArray['value']) as $optionId) {
-                if (isset($optionsArray['delete']) && $optionsArray['delete'][$optionId] == 1) {
-                    unset($optionsArray['value'][$optionId]);
+                if (isset($optionsArray['delete']) && isset($optionsArray['delete'][$optionId])) {
+                    if ($optionsArray['delete'][$optionId]== 1) {
```

Comment:
Exactly, and this is more effective than initial condition
```
if (isset($optionsArray['delete']) && isset($optionsArray['delete'][$optionId]) && $optionsArray['delete'][$optionId] == 1)
```
and much shorter ;)